server/commands.go

```diff
@@ -0,0 +1,472 @@
+	data.ApprovedQuestions[args.TeamId][args.ChannelId] = Extend(data.ApprovedQuestions[args.TeamId][args.ChannelId], question)
+	//from https://stackoverflow.com/a/37335777/199513
+	data.ProposedQuestions[args.TeamId][args.ChannelId] = append(data.ProposedQuestions[args.TeamId][args.ChannelId][:index], data.ProposedQuestions[args.TeamId][args.ChannelId][index+1:]...)
+
+	p.WriteToStorage(&data)
+
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         fmt.Sprintf("Question has been approved: %s", question),
+	}
+}
+
+func (p *Plugin) executeCommandIcebreakerShowProposals(args *model.CommandArgs) *model.CommandResponse {
+	sourceUser, _ := p.API.GetUser(args.UserId)
+	if !sourceUser.IsSystemAdmin() { //TODO: Check for Channel owner instead of System Admin
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "You need to be admin in order to show proposed questions",
+		}
+	}
+
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	if len(data.ProposedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no proposed questions for this channel...",
+		}
+	}
+
+	message := "Proposed questions:\n"
+	for index, question := range data.ProposedQuestions[args.TeamId][args.ChannelId] {
+		message = message + fmt.Sprintf("%d.\t%s:\t%s\n", index, question.Creator, question.Question)
+	}
+
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         message,
+	}
+}
+
+func (p *Plugin) executeCommandIcebreakerShowApproved(args *model.CommandArgs) *model.CommandResponse {
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	if len(data.ApprovedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no questions for this channel...",
+		}
+	}
+
+	message := "Questions:\n"
+	for index, question := range data.ApprovedQuestions[args.TeamId][args.ChannelId] {
+		message = message + fmt.Sprintf("%d.\t%s:\t%s\n", index, question.Creator, question.Question)
+	}
+
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         message,
+	}
+}
+
+func (p *Plugin) executeCommandIcebreaker(args *model.CommandArgs) *model.CommandResponse {
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	//check if there are any approved questions yet
+	if len(data.ApprovedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no approved questions that I can ask...",
+		}
+	}
+
+	//get a random user that is not a bot
+	user, err := p.GetRandomUser(args.ChannelId)
+	if err != nil {
+		return &model.CommandResponse{
```

Comment:
The error isn't always due to a lack of users. There can be multiple reasons of error. Currently, the user wouldn't know if there's something wrong or is the plugin not picking the data right. Maybe we can let the user know that there was an error?